File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int expected = 0, actual = 0;

        for (int i = 0; i < nums.size(); i++)
        {
            expected += i;
            actual += nums[i];
        }

        expected += nums.size();

        return expected - actual;
    }
};

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int targetSum = 0;
        int realSum = 0;

        for (int i = 0; i < nums.length; i++)
        {
            realSum += nums[i];
            targetSum += i;
        }
        targetSum += nums.length;
        return targetSum - realSum;
    }
}